fortitude/src/fix/snippet.rs

```diff
@@ -0,0 +1,45 @@
+#![allow(dead_code)]
+// Taken from ruff
+// Copyright 2022 Charles Marsh
+// SPDX-License-Identifier: MIT
+
+use unicode_width::UnicodeWidthStr;
+
+/// A snippet of source code for user-facing display, as in a diagnostic.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct SourceCodeSnippet(String);
```

Comment:
This isn't used anywhere yet, but potentially useful -- instead of storing a `String` in a `Violation`, store one of these, and it will be automatically truncated if it's too long.